CMakeLists.txt

```diff
@@ -133,11 +148,28 @@ include_directories(${LIBUSB_INCLUDE_DIR})
 include_directories(include)
 include_directories(src/mingw)
 
-add_library(${PROJECT_NAME} STATIC
+add_library(${PROJECT_NAME} SHARED
             ${STLINK_HEADERS} # header files for ide projects generated by cmake
             ${STLINK_SOURCE})
 target_link_libraries(${PROJECT_NAME} ${LIBUSB_LIBRARY})
 
+if (WIN32 OR MSYS OR MINGW)
+	set(STLINK_SHARED_VERSION
+		${STLINK_PACKAGE_VERSION_MAJOR}.${STLINK_PACKAGE_VERSION_MINOR})
+else()
+	set(STLINK_SHARED_VERSION
+		${STLINK_PACKAGE_VERSION_MAJOR}.${STLINK_PACKAGE_VERSION_MINOR}.${STLINK_PACKAGE_VERSION_PATCH})
+endif()
+
+set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${STLINK_PACKAGE_VERSION_MAJOR}
+VERSION ${STLINK_SHARED_VERSION})
+
+add_library(${PROJECT_NAME}static STATIC
```

Comment:
Please separate `${PROJECT_NAME}static` with a `-`. Maybe better would be to have a variable like `STLINK_LIB_STATIC`, `STLINK_LIB_SHARED` for the target names.